posts/2019/11/Variable-Explorer.md

```diff
@@ -0,0 +1,73 @@
+
+These include support to inspect any kind of Python object through a new viewer, the ability to filter and search for variables by name and type, and support for dataframes whose indexes have multiple dimensions.
+
+It is important to mention that some of the above improvements were possible through the integration of code coming from two different projects into our own. The first of them is called [gtabview](https://github.com/TabViewer/gtabview), whose code was used for the implementation of multi-dimensional Pandas indexes. And the second one is [objbrowser](https://github.com/titusjan/objbrowser), used for viewing arbitrary Python objects.
+
+<!-- TEASER_END -->
+
+
+## New viewer for arbitrary Python objects
+
+
+For Spyder 4 we added a long-requested functionality: inspecting any kind of Python object through the Variable Explorer. For a long time Spyder has had very good support to view and edit a small subset of Python objects: Numpy arrays, Pandas dataframes and series, and collections (i.e. lists, dictionaries and tuples). Other objects were displayed in a table-like view, and inspecting any of their attributes required showing a new table,
+
+![Python Viewer](/images/spyder-variable-explorer/python-viewer.png)
+
+This made very cumbersome to use this functionality, which was the reason why arbitrary Python objects were hidden by default from the Variable Explorer view.  
+In this version we decided to integrate the excellent [objbrowser](https://github.com/titusjan/objbrowser) project, by Pepijn Kenter ([@titusjan](https://github.com/titusjan)), which provides a tree-like view of Python objects, for a much simpler and comfortable way to inspect them.
+
+![Python Viewer Metadata](/images/spyder-variable-explorer/python-viewer-metadata.png)
+
+As can be seen above, this viewer will also allow users to have access to some extra metadata about the inspected object, such as its documentation, source code and the file that holds it.  
+It is very important to note that this work was accomplished thanks to the generosity of Pepijn, who kindly changed the license of [objbrowser](https://github.com/titusjan/objbrowser) to allow us integrating it with Spyder.
+To expose this new functionality we also decided to turn off the Variable Explorer option called “Exclude unsupported types” and introduce a new one called “Exclude callables and modules” (turned on by default) to display a much larger fraction of objects after a file or cell execution.
+
+![Exclude callables modules](/images/spyder-variable-explorer/exclude-callables-modules.png)
+
+Finally, we added a context-menu action to allow using the object explorer viewer for all objects, regardless if they have a default viewer or not.
```

Comment:
```suggestion
Finally, we added a context-menu action to open any object using the new Object Explorer even if they already have a builtin viewer (DataFrames, arrays, etc), allowing for deeper inspection of the inner workings of these datatypes.
```

- Clarify syntax
- Describe more clearly the benefit to the enhancement (at least as I originally proposed it)